lib/player.js

```diff
@@ -724,6 +736,73 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     if (mediaElement) {
       this.attach(mediaElement, /* initializeMediaSource= */ true);
     }
+
+    if (canvas) {
+      this.attachCanvas(canvas);
+    }
+  }
+
+  /**
+   * Create a shaka.lcevc.Dil object
+   * @param {LcevcDil.LcevcDilConfig} config
+   * @private
+   */
+  createDIL_(config) {
+    if (this.lcevcDil_ == null) {
+      this.lcevcDil_ = new shaka.lcevc.Dil(
+          /** @type {HTMLVideoElement} */ (this.video_),
+          this.lcevcCanvas_,
+          config,
+      );
+      if (this.mediaSourceEngine_) {
+        this.mediaSourceEngine_.updateLcevcDil(this.lcevcDil_);
+      }
+    }
+  }
+
+  /**
+   * Close a shaka.lcevc.Dil object if present and hide the canvas.
+   * @private
+   */
+  closeDIL_() {
+    if (this.lcevcDil_ != null) {
+      this.lcevcDil_.release();
+      this.lcevcDil_ = null;
+    }
+  }
+
+  /**
+   * Setup shaka.lcevc.Dil object
+   * @private
+   */
+  setupLcevc_(config) {
+    const safariVersion = shaka.util.Platform.safariVersion();
+    if (safariVersion) {
+      if (config.streaming.useNativeHlsOnSafari) {
+        shaka.log.alwaysWarn('LCEVC Error: Only Safari with MSE enabled ' +
+          'is supported. Please set config.streaming.useNativeHlsOnSafari ' +
+          'to false for LCEVC decoding.' );
+        return;
+      }
+    }
+
+    const tracks = this.getVariantTracks();
+    if (tracks && tracks[0] &&
+      tracks[0].videoMimeType ==
+      shaka.Player.SRC_EQUAL_EXTENSIONS_TO_MIME_TYPES_['ts']) {
+      const edge = shaka.util.Platform.isEdge() ||
+        shaka.util.Platform.isLegacyEdge();
+      if (edge) {
+        if (!config.streaming.forceTransmuxTS) {
+          // Warning for MPEG-2 TS LCEVC data is not passed to lib unless
+          // forceTransmux is enabled.
+          shaka.log.alwaysWarn('LCEVC Warning: For MPEG-2 TS decoding '+
+          'the config.streaming.forceTransmux must be enabled.');
```

Comment:
You defined `shaka.lcevc.Dil.ContainerFormat.MPEG2_TS`.  Are you sure transmuxing TS to MP4 is required?

Reply:
For Edge with TS contents, it somehow strips the LCEVC content off, if this parameter is not set. Hence the check.

Reply:
This would make a really great code comment.  A "why" comment is always more useful than a "what" comment, especially when you know something other people don't.